alvr/server/cpp/platform/linux/EncodePipelineNvEnc.cpp

```diff
@@ -0,0 +1,111 @@
+#include "EncodePipelineNvEnc.h"
+#include "ALVR-common/packet_types.h"
+#include "alvr_server/Settings.h"
+#include "ffmpeg_helper.h"
+#include <chrono>
+
+extern "C" {
+#include <libavcodec/avcodec.h>
+#include <libavutil/opt.h>
+#include <libswscale/swscale.h>
+}
+
+namespace {
+
+const char *encoder(ALVR_CODEC codec) {
+    switch (codec) {
+    case ALVR_CODEC_H264:
+        return "h264_nvenc";
+    case ALVR_CODEC_H265:
+        return "hevc_nvenc";
+    }
+    throw std::runtime_error("invalid codec " + std::to_string(codec));
+}
+
+} // namespace
+alvr::EncodePipelineNvEnc::EncodePipelineNvEnc(std::vector<VkFrame> &input_frames,
+                                               VkFrameCtx &vk_frame_ctx) {
+    auto input_frame_ctx = (AVHWFramesContext *)vk_frame_ctx.ctx->data;
+    assert(input_frame_ctx->sw_format == AV_PIX_FMT_BGRA);
+
+    int err;
+    for (auto &input_frame : input_frames) {
+        vk_frames.push_back(input_frame.make_av_frame(vk_frame_ctx).release());
```

Comment:
Is there a specific reason to go from managed pointers to raw pointers?

Reply:
it's managed

Reply:
I meant that as in https://en.cppreference.com/book/intro/smart_pointers

I dislike writing destructors, and the unique_ptr is safer if in the future we allow replacing the frames, or any runtime configuration that could invalidate them.

Reply:
I see, yeah will change

Reply:
Waiting on this to merge.

Reply:
done